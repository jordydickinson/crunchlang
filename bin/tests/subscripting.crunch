fun subscript_get(): int32 {
    let xs = {1, 2, 3};
    return xs[1];
}

fun subscript_set!() {
    var xs = {1, 2, 3};
    xs[1] := 4;
}
