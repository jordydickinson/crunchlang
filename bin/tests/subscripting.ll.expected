; ModuleID = 'subscripting'
source_filename = "subscripting"

@llvm.global_ctors = appending global [1 x void ()*] [void ()* @init.ctors]

define i32 @"subscript_get$pure"() {
entry:
  br label %body

body:                                             ; preds = %entry
  %xs = alloca [3 x i32]
  store [3 x i32] [i32 1, i32 2, i32 3], [3 x i32]* %xs
  %xs.i = getelementptr [3 x i32], [3 x i32]* %xs, i32 0, i32 1
  %xs.i.0 = load i32, i32* %xs.i
  ret i32 %xs.i.0
}

define void @subscript_set() {
entry:
  br label %body

body:                                             ; preds = %entry
  %xs = alloca [3 x i32]
  store [3 x i32] [i32 1, i32 2, i32 3], [3 x i32]* %xs
  %xs.i = getelementptr [3 x i32], [3 x i32]* %xs, i32 0, i32 1
  store i32 4, i32* %xs.i
  ret void
}

define void @init.ctors() {
entry:
  ret void
}
